packages/dd-trace/src/appsec/api_security_sampler.js

```diff
@@ -1,61 +1,87 @@
 'use strict'
 
+const LRUCache = require('lru-cache')
+const web = require('../plugins/util/web')
 const log = require('../log')
+const { AUTO_REJECT, USER_REJECT } = require('../../../../ext/priority')
 
-let enabled
-let requestSampling
+const MAX_SIZE = 4096
+const DEFAULT_DELAY = 30 // 30s
 
-const sampledRequests = new WeakSet()
+let enabled
+let sampledRequests
 
 function configure ({ apiSecurity }) {
   enabled = apiSecurity.enabled
-  setRequestSampling(apiSecurity.requestSampling)
+  const delay = apiSecurity.sampleDelay || DEFAULT_DELAY
+  sampledRequests = new LRUCache({ max: MAX_SIZE, ttl: delay * 1000 })
```

Comment:
I agree that `lru-cache` doesn’t have strong TTL guarantees, which is why I implemented it from scratch instead of using `@isaacs/ttlcache` (I chose to go with the custom approach).
IMHO I can create a separate PR to switch to `@isaacs/ttlcache` and update all the relevant areas such as https://github.com/DataDog/dd-trace-js/blob/ba53c01cf2bee861028fc5bc96ef292de3e3bbd6/packages/dd-trace/src/debugger/devtools_client/status.js#L21

Reply:
wdym you implemented it from scratch ? aren't you just using the lru-cache module ?